app/src/main/java/com/example/user/epam_hw/MainActivity.java

```diff
@@ -1,23 +1,63 @@
 package com.example.user.epam_hw;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.widget.Toast;
+import android.support.design.widget.NavigationView;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.app.AppCompatActivity;
+import android.view.MenuItem;
+
 public class MainActivity extends AppCompatActivity {
 
+    private FirstFragment firstFragment;
```

Comment:
do not save instance of any fragments! only if you are using some special logic with retained state it is possible to keep it.